crates/router/src/connector/utils.rs

```diff
@@ -1980,6 +1980,19 @@ impl
     }
 }
 
+pub fn get_card_details(
+    payment_method_data: domain::PaymentMethodData,
+    connector_name: &'static str,
+) -> Result<domain::payments::Card, errors::ConnectorError> {
+    match payment_method_data {
+        domain::PaymentMethodData::Card(details) => Ok(details),
+        _ => Err(errors::ConnectorError::NotSupported {
+            message: SELECTED_PAYMENT_METHOD.to_string(),
+            connector: connector_name,
```

Comment:
Is there a scope in future to add other payment methods into the current implemented flows?

Reply:
No. external 3ds is only available for cards.